src/libraries/System.Private.Uri/src/System/UriExt.cs

```diff
@@ -309,6 +332,16 @@ public static bool TryCreate(Uri? baseUri, Uri? relativeUri, [NotNullWhen(true)]
         }
 
         public string GetComponents(UriComponents components, UriFormat format)
+        {
+            if (DisablePathAndQueryCanonicalization && (components & (UriComponents.Path | UriComponents.Query)) != 0)
+            {
+                throw new InvalidOperationException(SR.net_uri_GetComponentsCalledWhenCanonicalizationDisabled);
+            }
```

Comment:
Offline discussion: I'm concerned that there's no way for the consumer to know if DisablePathAndQueryCanonicalization is set and avoid this exception. This method can also throw IOE for relative Uris, but there you can check IsAbsoluteUri first.

Reply:
> This method can also throw IOE for relative Uris, but there you can check IsAbsoluteUri first.

In the current version this is unreachable for relative Uris (since `UriKind` hasn't been exposed yet), but I will keep it in mind when adding the rest of the API. It's not critical as it would always throw anyway, but the change wasn't intentional.

Reply:
I think we should consider adding a way to expose this, something like IsPathAndQueryCanonicalizationDisabled.

Reply:
Just to be super-clear: We can track that for 7.0, we do not consider it blocking attempt for 6.0 backport.
@geoffkizer please let us know if you disagree.